src/e2e.js

```diff
@@ -150,14 +155,8 @@ mye2e.prototype.signEncrypt = function(data, encryptKey, sign) {
     } else {
       signKey = null;
     }
-    return new Promise(
-      function(resolve, reject) {
-        pgp.encryptSign(buf2array(data), [], keys, [], signKey).addCallback(
-          function (ciphertext) {
-            resolve(array2buf(ciphertext));
-          }).addErrback(reject);
-      });
-  });
+    return pgp.encryptSign(buf2array(data), [], keys, [], signKey);
+  }).then(array2buf);
```

Comment:
Nice! I imagine a similar streamlining could be done for verifyDecrypt - if you want to change it go for it, else I'll do it after this pull request.